firefox_puppeteer/ui/tabbar.py

```diff
@@ -3,126 +3,315 @@
-                               .find_element('anon attribute',
-                                             {'anonid': 'tabs-newtab-button'}))
+        return MenuPanel(lambda: self.marionette, self.window)
 
     @property
-    def active_tab(self):
+    def newtab_button(self):
+        """The DOM element which represents the new tab button.
+
+        :returns: Reference to the new tab button
         """
-        :returns: The :class:`TabElement` corresponding to the currently active
-                  tab.
+        return self.toolbar.find_element('anon attribute', {'anonid': 'tabs-newtab-button'})
+
+    @property
+    def tabs(self):
+        """List of all the :class:`Tab` instances of the current browser window.
+
+        :returns: List of :class:`Tab`'s
         """
-        for tab in self.tabs:
-            if tab.is_active():
-                return tab
+        tabs = self.toolbar.find_elements('tag name', 'tab')
+
+        return [Tab(lambda: self.marionette, self.window, tab) for tab in tabs]
 
     @property
-    def menupanel(self):
+    def toolbar(self):
+        """The DOM element which represents the tab toolbar.
+
+        :returns: Reference to the tabs toolbar
         """
-        Provides access to the menu popup. This is the menu opened after
-        clicking the settings button on the right hand side of the browser.
+        return self.marionette.find_element('id', 'tabbrowser-tabs')
+
+    # Properties for helpers when working with the tabs toolbar #
 
-        See the :class:`~ui.menu.MenuPanel` reference.
+    @property
+    def selected_index(self):
+        """The index of the currently selected tab.
+
+        :return: Index of the selected tab
         """
-        return MenuPanel(lambda: self.marionette)
+        return int(self.toolbar.get_attribute('selectedIndex'))
 
     @property
-    def tabs(self):
+    def selected_tab(self):
+        """A :class:`Tab` instance of the currently selected tab.
+
+        :returns: :class:`Tab` instance
         """
-        :returns: A list of all the :class:`TabElement`'s.
+        return self.tabs[self.selected_index]
+
+    # Methods for helpers when working with the tabs toolbar #
+
+    def close_all_tabs(self, exceptions=None):
+        """Forces closing of all open tabs.
+
+        There is an optional `exceptions` list, which can be used to exclude
+        specific tabs from being closed.
+
+        :param exceptions: Optional, list of :class:`Tab` instances not to close
         """
-        tabs = (self.marionette.find_element('id', 'tabbrowser-tabs')
-                               .find_elements('tag name', 'tab'))
-        return [self.TabElement(tab) for tab in tabs]
+        # Get handles from tab exceptions, and find those which can be closed
+        for tab in self.tabs:
+            if tab not in exceptions:
+                tab.close(force=True)
+
+    def close_tab(self, tab=None, *args, **kwargs):
```

Comment:
I just wanted to avoid the redefinition of the default values. But lets change it. Not sure if there is a better way for it or not.